### PR TITLE
[docker-compose] listen for dc events, update hud with dc status [ch977]

### PR DESCRIPTION
**Pull request by maiamcc**
include a hacky map from dc event --> probable status.

no real UI thought given to this yet. Looks like this:
dc hud

### CODE REVIEW

File: internal/dockercompose/event.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	err := json.Unmarshal(b, &evt)
+
+	return evt, err
+}
+
+// https://docs.docker.com/engine/reference/commandline/events/
+type Type string
+
+const (
+	// Add 'types' here (and to `UnmarshalJSON` below`) as we support them
```
Comment (jazzdan) [inline internal/dockercompose/event.go]: I think you accidentally a ` here :P

File: internal/engine/docker_compose_watch.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	"github.com/windmilleng/tilt/internal/logger"
+	"github.com/windmilleng/tilt/internal/store"
+)
+
+type DockerComposeWatcher struct {
+	watching bool
+}
+
+func NewDockerComposeWatcher() *DockerComposeWatcher {
+	fmt.Println("making a docker-compose watcher hoorayyyy")
```
Comment (jazzdan) [inline internal/engine/docker_compose_watch.go]: 👀

File: internal/engine/upper.go [hunk 3]
```diff
@@ -649,6 +653,24 @@ func handleExitAction(state *store.EngineState, action hud.ExitAction) {
 	}
 }
 
+func handleDockerComposeEvent(ctx context.Context, engineState *store.EngineState, action DockerComposeEventAction) {
+	evt := action.Event
+	mn := evt.Service
+	ms, ok := engineState.ManifestStates[model.ManifestName(mn)]
+	if !ok {
+		// No corresponding manifest, nothing to do
+		logger.Get(ctx).Infof("event for unrecognized manifest %s", mn)
```
Comment (jazzdan) [inline internal/engine/upper.go]: Should you `return` in this case?

File: internal/hud/view/view.go [hunk 4]
```diff
@@ -31,16 +31,22 @@ type Resource struct {
...
+	// Relevant to k8s resources (maybe should accomplish via interface?)
 	PodName         string
 	PodCreationTime time.Time
 	PodStatus       string
 	PodRestarts     int
 	Endpoints       []string
-	PodLog          string
+
+	// Relevant to docker-compose resources (maybe should accomplish via interface?)
+	IsDCManifest bool
```
Comment (jazzdan) [inline internal/hud/view/view.go]: Again I would advocate for making this a separate type, but we previously discussed this and you're foregoing that for now.

File: internal/engine/docker_compose_watch.go (hunk 2 above)
Comment (maiamcc) [reply internal/engine/docker_compose_watch.go]: 👀

File: internal/engine/upper.go (hunk 3 above)
Comment (maiamcc) [reply internal/engine/upper.go]: lol yes 😳

**Pull request merged**